### PR TITLE
Permanent option

### PR DESCRIPTION
**Pull request by breyell**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "mechanic",
-  "version": "1.4.0",
+  "version": "1.4.1",
```
Comment (abea) [inline package.json:3]: New minor version, right?

Comment (breyell) [reply package.json:3]: Honestly haven't had to worry to worry about semver ever so no idea. 😬

Comment (abea) [reply package.json:3]: I think so since it adds a feature.

**Comment by boutell** [comment]
Fixes bug: bump patchlevel (third number).
Introduces feature: bump minor version (second number).
BC break: bump major version (after consultation that this is even a necessary thing to be doin').

File: template.conf [hunk 2]
```diff
@@ -76,11 +76,11 @@
         {% if site.autoindex %}
           autoindex on;
           {% if site.backends.length %}
-            try_files $uri $uri/ @proxy-{{ site.shortname }}-{{ options.port }};
+            try_files $uri $uri/ @proxy-{{ site.shortname }}-{{ options.port }}{% if options.permanent %} permanent{% endif %};
           {% endif %}
         {% else %}
           {% if site.backends.length %}
-            try_files $uri @proxy-{{ site.shortname }}-{{ options.port }};
+            try_files $uri @proxy-{{ site.shortname }}-{{ options.port }}{% if options.permanent %} permanent{% endif %};
```
Comment (boutell) [inline template.conf:83]: This is a try_files directive, not a redirect. I don't think permanent would work here. I think you meant to do this elsewhere, where the various redirect options are handled. Also this needs to be tested on an actual server.

**Review by boutell — CHANGES_REQUESTED** [review]
I would encourage you to set up nginx on your local machine and experiment with self-issued certs and /etc/hosts edits to point names to localhost, so that you can fully test this locally; or else you could do it with a server.

**Comment by abea** [comment]
I'm going to leave it to the expert to do code review on this one.

File: template.conf (hunk 2 above)
Comment (breyell) [reply template.conf:83]: https://www.nginx.com/blog/creating-nginx-rewrite-rules/#Examples&nbsp;%E2%80%93-Standardizing-the-Domain-Name

Comment (breyell) [reply template.conf:83]: It's not the recommended way to solve but it is a way. 🤷🏻‍♂️

Comment (breyell) [reply template.conf:83]: You're right! I meant to put this at the end of a `rewrite`. I shall fix.

**Comment by breyell** [comment]
Still need to test on a server but moved `permanent` to the right location for now.

**Comment by boutell** [comment]
Bump, @breyell did this ever get tested?

**Comment by breyell** [comment]
Unfortunately, no. =\

**Comment by breyell** [comment]
@boutell This has now been tested! Tweaked `options.permanent` to `site.permanent`.

**Review by boutell — APPROVED** [review]

**Pull request merged**